workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/executor/request/client/HttpClient.java

```diff
@@ -29,13 +34,28 @@ public Response execute(String method, String url, Object body, Map<String, Stri
     int responseCode = httpResponse.getCode();
 
     if (classicHttpResponse.getEntity() != null && classicHttpResponse.getEntity().getContent() != null) {
-      return new Response(responseCode,
-          IOUtils.toString(classicHttpResponse.getEntity().getContent(), StandardCharsets.UTF_8));
+      return this.handleResponse(responseCode,
+          IOUtils.toString(classicHttpResponse.getEntity().getContent(), StandardCharsets.UTF_8),
+          classicHttpResponse.getFirstHeader(HttpHeaders.CONTENT_TYPE));
     } else {
       return new Response(responseCode, "");
     }
   }
 
+  private Response handleResponse(int statusCode, String content, Header contentType) {
+    Object data = content;
+    if (contentType == null || (contentType.getName().equals(HttpHeaders.CONTENT_TYPE) && contentType.getValue()
```

Comment:
we can introduce a isJsonContent() private method or something to help read the complex condition